### PR TITLE
fix(build): fixes React-RCTText build with RN 0.69.0

### LINKED ISSUE #33976: Error :  Use of undeclared identifier 'YGValue' when building iOS
### Description

Hello,

I have the following error when building with the latest RN version.

I was just trying to upgrade... 
iScreen Shoter - 2022-06-08 10:29:49 688

```
Error :  Use of undeclared identifier 'YGValue' when building iOS
```

### Version

0.69-RC.6

### Output of `npx react-native info`

System:
    OS: macOS 12.2.1
    CPU: (10) x64 Apple M1 Pro
    Memory: 1.71 GB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.19.1 - /usr/local/bin/node
    Yarn: 1.22.18 - /opt/homebrew/bin/yarn
    npm: 6.14.16 - /usr/local/bin/npm
    Watchman: 2022.03.21.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.4, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK: Not Found
  IDEs:
    Android Studio: 2021.1 AI-211.7628.21.2111.8309675
    Xcode: 13.3.1/13E500a - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.15 - /Library/Java/JavaVirtualMachines/zulu-11.jdk/Contents/Home/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.0.0 => 18.0.0
    react-native: 0.69.0-rc.6 => 0.69.0-rc.6
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

I tried to upgrade...

### Snack, code example, screenshot, or link to a repository

N/A

### PR DESCRIPTION
**Pull request by ph4r05**
## Summary

Fixes iOS build for React-RCTText with RN 0.69.0, fixes #33976

PR contains changes from https://github.com/facebook/react-native/issues/33976#issuecomment-1164973209

PoC repo: https://github.com/ph4r05/poc-rn-34064

Related issues:
- https://github.com/expo/expo/issues/16283
- https://github.com/facebook/react-native/issues/33815
- https://github.com/facebook/react-native/issues/33976

## Changelog

[iOS] [Fixed] - Fix build for React-RCTText

### CODE REVIEW

**Comment by renchap** [comment]
I can confirm it fixes things for me, but I am not sure why the `.m` files would not be picked by the compiler anymore.

Also this seems to only occur when using some third party libraries (`react-native-safe-area-context` ?).

**Review by cortinico — APPROVED** [review]
Thanks for sending this over @ph4r05 
We're going to merge this as it was confirmed as valid by @renchap here https://github.com/facebook/react-native/issues/33976#issuecomment-1165360882

I'll defer to @cipolleschi for a follow-up on the iOS side if we need one.

**Comment by cortinico** [comment]
> I can confirm it fixes things for me, but I am not sure why the `.m` files would not be picked by the compiler anymore.
> Also this seems to only occur when using some third party libraries (`react-native-safe-area-context` ?).

Also as an heads up, this is currently failing to build RN Tester internally with:

```
CompileC [...]/Library/Developer/Xcode/DerivedData/RNTesterPods-avhcctdswdycllfjzyasqkbrubnn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/React-RCTText.build/Objects-normal/x86_64/RCTInputAccessoryShadowView.o /[...]/Libraries/Text/TextInput/RCTInputAccessoryShadowView.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler (in target 'React-RCTText' from project 'Pods')
    cd /[...]/packages/rn-tester/Pods
    export LANG\=en_US.US-ASCII
    /usr/local/fbprojects/packages/xcode/217/xcode_13.3.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c -target x86_64-apple-ios12.4-simulator -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -std\=gnu11 -fobjc-arc -fmodules -fmodules-cache-path\=/[...]/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -fmodules-prune-interval\=86400 -fmodules-prune-after\=345600 -fbuild-session-file\=/[...]/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror\=non-modular-include-in-framework-module -Wno-trigraphs -fpascal-strings -O0 -fno-common -Wno-missing-field-initializers -Wno-missing-prototypes -Werror\=return-type -Wdocumentation -Wunreachable-code -Wno-implicit-atomic-properties -Werror\=deprecated-objc-isa-usage -Wno-objc-interface-ivars -Werror\=objc-root-class -Wno-arc-repeated-use-of-weak -Wimplicit-retain-self -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wconditional-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wno-float-conversion -Wnon-literal-null-conversion -Wobjc-literal-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wundeclared-selector -Wdeprecated-implementations -DPOD_CONFIGURATION_DEBUG\=1 -DDEBUG\=1 -DCOCOAPODS\=1 -DOBJC_OLD_DISPATCH_PROTOTYPES\=0 -isysroot /usr/local/fbprojects/packages/xcode/217/xcode_13.3.1.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator15.4.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -g -Wno-sign-conversion -Winfinite-recursion -Wcomma -Wblock-capture-autoreleasing -Wstrict-prototypes -Wno-semicolon-before-method-body -Wunguarded-availability -fobjc-abi-version\=2 -fobjc-legacy-dispatch -index-store-path /[...]/Library/Developer/Xcode/DerivedData/RNTesterPods-avhcctdswdycllfjzyasqkbrubnn/Index/DataStore -iquote /[...]/Library/Developer/Xcode/DerivedData/RNTesterPods-avhcctdswdycllfjzyasqkbrubnn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/React-RCTText.build/React-RCTText-generated-files.hmap -I/[...]/Library/Developer/Xcode/DerivedData/RNTesterPods-avhcctdswdycllfjzyasqkbrubnn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/React-RCTText.build/React-RCTText-own-target-headers.hmap -I/[...]/Library/Developer/Xcode/DerivedData/RNTesterPods-avhcctdswdycllfjzyasqkbrubnn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/React-RCTText.build/React-RCTText-all-non-framework-target-headers.hmap -ivfsoverlay /[...]/Library/Developer/Xcode/DerivedData/RNTesterPods-avhcctdswdycllfjzyasqkbrubnn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/React-RCTText.build/all-product-headers.yaml -iquote /[...]/Library/Developer/Xcode/DerivedData/RNTesterPods-avhcctdswdycllfjzyasqkbrubnn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/React-RCTText.build/React-RCTText-project-headers.hmap -I/[...]/Library/Developer/Xcode/DerivedData/RNTesterPods-avhcctdswdycllfjzyasqkbrubnn/Build/Products/Debug-iphonesimulator/React-RCTText/include -I/[...]/packages/rn-tester/Pods/Headers/Private -I/[...]/packages/rn-tester/Pods/Headers/Private/React-RCTText -I/[...]/packages/rn-tester/Pods/Headers/Public -I/[...]/packages/rn-tester/Pods/Headers/Public/DoubleConversion -I/[...]/packages/rn-tester/Pods/Headers/Public/RCT-Folly -I/[...]/packages/rn-tester/Pods/Headers/Public/React-Core -I/[...]/packages/rn-tester/Pods/Headers/Public/React-RCTText -I/[...]/packages/rn-tester/Pods/Headers/Public/React-callinvoker -I/[...]/packages/rn-tester/Pods/Headers/Public/React-cxxreact -I/[...]/packages/rn-tester/Pods/Headers/Public/React-jsi -I/[...]/packages/rn-tester/Pods/Headers/Public/React-jsiexecutor -I/[...]/packages/rn-tester/Pods/Headers/Public/React-jsinspector -I/[...]/packages/rn-tester/Pods/Headers/Public/React-logger -I/[...]/packages/rn-tester/Pods/Headers/Public/React-perflogger -I/[...]/packages/rn-tester/Pods/Headers/Public/React-runtimeexecutor -I/[...]/packages/rn-tester/Pods/Headers/Public/Yoga -I/[...]/packages/rn-tester/Pods/Headers/Public/fmt -I/[...]/packages/rn-tester/Pods/Headers/Public/glog -I/[...]/Library/Developer/Xcode/DerivedData/RNTesterPods-avhcctdswdycllfjzyasqkbrubnn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/React-RCTText.build/DerivedSources-normal/x86_64 -I/[...]/Library/Developer/Xcode/DerivedData/RNTesterPods-avhcctdswdycllfjzyasqkbrubnn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/React-RCTText.build/DerivedSources/x86_64 -I/[...]/Library/Developer/Xcode/DerivedData/RNTesterPods-avhcctdswdycllfjzyasqkbrubnn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/React-RCTText.build/DerivedSources -F/[...]/Library/Developer/Xcode/DerivedData/RNTesterPods-avhcctdswdycllfjzyasqkbrubnn/Build/Products/Debug-iphonesimulator/React-RCTText -fmodule-map-file\=/[...]/packages/rn-tester/Pods/Headers/Public/React/React-Core.modulemap -fmodule-map-file\=/[...]/packages/rn-tester/Pods/Headers/Public/folly/RCT-Folly.modulemap -fmodule-map-file\=/[...]/packages/rn-tester/Pods/Headers/Public/yoga/Yoga.modulemap -include /[...]/packages/rn-tester/Pods/Target\ Support\ Files/React-RCTText/React-RCTText-prefix.pch -MMD -MT dependencies -MF /[...]/Library/Developer/Xcode/DerivedData/RNTesterPods-avhcctdswdycllfjzyasqkbrubnn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/React-RCTText.build/Objects-normal/x86_64/RCTInputAccessoryShadowView.d --serialize-diagnostics /[...]/Library/Developer/Xcode/DerivedData/RNTesterPods-avhcctdswdycllfjzyasqkbrubnn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/React-RCTText.build/Objects-normal/x86_64/RCTInputAccessoryShadowView.dia -c /[...]/Libraries/Text/TextInput/RCTInputAccessoryShadowView.m -o /[...]/Library/Developer/Xcode/DerivedData/RNTesterPods-avhcctdswdycllfjzyasqkbrubnn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/React-RCTText.build/Objects-normal/x86_64/RCTInputAccessoryShadowView.o -index-unit-output-path /Pods.build/Debug-iphonesimulator/React-RCTText.build/Objects-normal/x86_64/RCTInputAccessoryShadowView.o
error: Build input file cannot be found: '/[...]/Libraries/Text/TextInput/RCTInputAccessoryShadowView.m' (in target 'React-RCTText' from project 'Pods')
```

**Comment by renchap** [comment]
> I just created an app with 0.69 and tried to build with both the legacy and the New Architecture and it builds without issues and without applying the patch.
> Is there some dependencies that trigger the issue?

Hi @cipolleschi,

Sorry for not yet providing a sample reproducing the issue, I am in the final steps of releasing an app and did not had time to investigate this as properly as I wanted.

This is most probably only occurring when a dependency is used. In the linked issue (#33976) we noticed that `react-native-safe-area-context` needed a similar patch, so maybe this is the dependency that triggers this?

**Comment by cipolleschi** [comment]
@renchap I tried to add the `react-native-safe-area-context` dependencies to a new app using `yarn add react-native-safe-area-context`. The dependency has been installed in the `node_modules`.

Then I tried both to install the pods with the new and the legacy architectures. In both cases i could see the pod in the project and the pod install script reporting this line:

```
Installing react-native-safe-area-context 4.3.1
```

In both case, after a clean, the app was building and running.

I don't think we need this fix at all. My best guess is that a step was missing when migrating the app.

**Comment by ph4r05** [comment]
@cipolleschi this is a minimal PoC that behaves as described in the issue: https://github.com/ph4r05/poc-rn-34064

**Comment by ph4r05** [comment]
@renchap pls which react-native version do you use in the project?

**Comment by cipolleschi** [comment]
RN 0.69.0

**Comment by renchap** [comment]
@ph4r05 RN 0.69.0.

But I am also using expo (for some Expo modules, not the whole thing), so this could come from Expo indeed.

**Comment by cipolleschi** [comment]
I'm trying out the PoC
1. first commit build and run (as expected, being the react-init command)
2. second commit build and run as well. 
3. third commit fails with the error reported in the PR.

I then tried to isolate which change may generate that:
- Adding specific version of Flipper (`flipper_post_install(installer)` is already called by `react_native_post_install` so I don't think we should invoke it explicitly) -> App builds and run
- bumping iOS to 15.2 -> App builds and run
- adding expo to the package.json (yarn install && pod install) -> App builds and run
- add `require_relative '../node_modules/expo/scripts/autolinking'` && pod install -> Build fails.

So, there is something in the `autolinking` file of Expo that generates the failure. I tried to look a little bit further into it and looks like that that file requires another file called `autolinking_manager` which, in turn requires some other local files from a `cocoapods` folder which monkey-patch some files from Cocoapods framework. 

So, from this investigation, it looks like that it's Expo that is not supporting 0.69 yet and I don't think we have to actually merge this into React Native.

@Kudo what do you think?

**Comment by Kudo** [comment]
for expo to integrate React-Core from swift code, we have some patches for React-Core. sometimes the clang compiler requires the imports to be in correct order. otherwise, it will import to wrong headers and have build errors.

my fix toward the problem is like that, maybe it's lower risk. could you check whether it works for you and makes sense to you?

```diff
diff --git a/node_modules/react-native/React/Views/RCTShadowView.h b/node_modules/react-native/React/Views/RCTShadowView.h
index 428d2eb..226672f 100644
--- a/node_modules/react-native/React/Views/RCTShadowView.h
+++ b/node_modules/react-native/React/Views/RCTShadowView.h
@@ -8,6 +8,8 @@
 #import <UIKit/UIKit.h>

 #import <React/RCTComponent.h>
+// Keeps RCTConvert.h here before yoga for clang module to generate correct header imports.
+#import <React/RCTConvert.h>
 #import <React/RCTLayout.h>
 #import <React/RCTRootView.h>
 #import <yoga/Yoga.h>
```

**Comment by ph4r05** [comment]
@Kudo this worked! Shall I change this PR so it reflects this change?

**Comment by Kudo** [comment]
@ph4r05 sure, please update the pr if the change makes sense to you.

**Comment by ph4r05** [comment]
done! Great catch with the order of imports @Kudo !

**Comment by Kudo** [comment]
thanks everyone here to fix the issue together 🔥

File: React/Views/RCTShadowView.h [hunk 1]
```diff
@@ -8,6 +8,8 @@
 #import <UIKit/UIKit.h>
 
 #import <React/RCTComponent.h>
+// Keeps RCTConvert.h here before yoga for clang module to generate correct header imports.
+#import <React/RCTConvert.h>
```
Comment (cipolleschi) [inline React/Views/RCTShadowView.h:12]: As far as I can see, `RCTConvert.h` also import `yoga/Yoga`, so we could also remove line 15, but I don't think it makes any harm to keep it.

Comment (ph4r05) [reply React/Views/RCTShadowView.h:12]: I can remove it, no problem. Should I?

Comment (cipolleschi) [reply React/Views/RCTShadowView.h:12]: I would do it:
1. if it breaks on CI
2. only locally for knowledge's sake

So, probably no... :P

**Pull request closed without merging**